src/empirical.jl

```diff
@@ -9,6 +8,7 @@ struct ECDF{T <: AbstractVector{<:Real}, W <: AbstractWeights{<:Real}}
 end
 
 function (ecdf::ECDF)(x::Real)
+    isnan(x) && return NaN
```

Comment:
This should use the element type of `ECDF` to to ensure type stability of e.g. `Float32`s and `Dual` so probably something like `T(NaN)` for `ECDF{T}`.

Reply:
Hmm, there are already other type instabilities present:

```julia
weightsum = evenweights ? length(ecdf.sorted_values) : sum(ecdf.weights)
```

I'm fine fixing these here or in a different PR

Reply:
Never mind, fixes were easy.  I'll push in a sec

Reply:
Can you also add tests, including `@inferred`?

Reply:
Never mind again, way harder to handle NaN in a type stable way than I thought!  I'm gonna punt on this and ask this to be merged as-is.

Reply:
The `weightsum` type instability is not visible outside of the function for `Float32` since `partialsum/weightsum` will return a `Float64`. So `NaN` is more correct than `oftype(T, NaN)`. Not sure about `Dual`, would need to try that.

But please add new tests.